### ISSUE TITLE
useSizes hook cause re-rendering infinitely

### ISSUE THREAD

**Issue by hakonxxx** [elrumordelaluz/reactour#462]
**Describe the bug**
useSizes hook cause re-rendering infinitely in some cases.
popover is shaking(keep updating it`s transform) too.
visit https://codesandbox.io/s/jovial-perlman-2uiiup

**To Reproduce**
steps:
```ts
[{
  selector: ".a", // .a is target, with height smaller than window.offsetHeight
  content: "infinitely scroll",
  highlightedSelectors: [".a", ".b"] // .b is a element, with height larger than window.offsetHeight
}]
```
useSizes hook:
```ts
  const handleResize = useCallback(() => {
    setdDimensions(
      getHighlightedRect(target, step?.highlightedSelectors, step?.bypassElem)
    )
  }, [target, step?.highlightedSelectors])

  useEffect(() => {
    handleResize()
    window.addEventListener('resize', handleResize)
    return () => window.removeEventListener('resize', handleResize)
  }, [target, step?.highlightedSelectors, refresher])

  useEffect(() => {
    const isInView = inView({
      ...dimensions,
      threshold: scrollOptions.inViewThreshold,
    })
    if (!isInView && target) {
      setTransition(true)
      smoothScroll(target, scrollOptions)
        .then(() => {
          if (!observing) setRefresher(Date.now())
        })
        .finally(() => {
          setTransition(false)
        })
    }
  }, [dimensions])
```
1. first, `smoothScroll` and update `transition`(transform of popover may change)
2. after scrolling, `.a` should be in view, `observing` is `false`, update `refresher`
3. `refresher` updated, so call `handleResize` which will change `dimensions`
4. `dimensions` changed, goto **step 1**:
4.1. `isInView` still is false, because height of `.a` is smaller than `window.offsetHeight`, and `.a` should be in view, and height of `.b` is larger than `window.offsetHeight`
4.2 call `smoothScroll`(redundant)
5. ...

**Expected behavior**
stop re-rendering if scrolled into target and dimensions not changed.

**Fix**
```ts
  const handleResize = useCallback(() => {
    const newDimensions = getHighlightedRect(target, step?.highlightedSelectors, step?.bypassElem)
    if (Object.entries(dimensions).some(([key, value]) => newDimensions[key] !== value)) {
        setdDimensions(newDimensions)
    }
  }, [target, step?.highlightedSelectors, dimensions])
```
or change implement of `inView`

**BTW**
`setdDimensions` should be `setDimensions` right?

**Comment by elrumordelaluz** [comment]
Hi @hakonxxx, thanks for open the _Issue_ and sorry to keep attention right now. Let me check all the info you are sharing and back to you.

**Comment by elrumordelaluz** [comment]
hey @hakonxxx your proposed fix seem good to me! Mind make a PR?

Thanks!

**Comment by tobiasmanroth** [comment]
This fix would be awesome, I'm encountering something similar when an element is partly outside of the browser window.

**Comment by elrumordelaluz** [comment]
Thank you @hakonxxx, the fix proposed by you should be available on `@reactour/tour@2.10.4`.
